src/main/java/org/beanpod/switchboard/dto/mapper/InputChannelMapper.java

```diff
@@ -5,16 +5,21 @@
 import org.beanpod.switchboard.entity.InputChannelEntity;
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
+import org.openapitools.model.InputChannelModel;
 
 @Mapper(
     componentModel = "spring",
     uses = {DecoderMapper.class, ChannelMapper.class})
 public interface InputChannelMapper {
-  @Mapping(target = "decoder", ignore = true)
+  @Mapping(target = "decoder", qualifiedByName = "toDecoderDTOShallow")
   InputChannelDTO toInputChannelDTO(InputChannelEntity inputChannelEntity);
 
   @Mapping(target = "decoder", ignore = true)
   Set<InputChannelDTO> toInputChannelDTOs(Set<InputChannelEntity> inputChannelEntities);
 
   InputChannelEntity toInputChannelEntity(InputChannelDTO inputChannelDTO);
+
+  @Mapping(target = "name", source = "channel.name")
+  @Mapping(target = "port", source = "channel.port")
+  InputChannelModel inputChannelDTOToInputChannelModel(InputChannelDTO inputChannelDTO);
```

Comment:
why explicit target and source, i think since same name, they should map without target and source

Reply:
Mapstruct won't automatically map `inputChannelDTO.channel.name` to `inputChannelModel.name`, it doesn't check properties of properties. So we need to explicitly map otherwise it will remain null.

Reply:
ah yes i see, im wondering why is is our DTO structure different from our model

Reply:
i think the model should reference the channel in the yaml

Reply:
That's on purpose, I decided not to expose the `ChannelEntity` to the users because I view it more as something inherent to our database structure. So instead of getting
```
inputChannel:{
  id: 1
  channel: {
   port: 80
  }
}

```
you just get
```
inputChannel:{
  id: 1
  port: 80
}
```
Essentially we reduce the `Channel` object into the `InputChannel` object

Reply:
i think that makes sense